crates/iota-metrics/src/hardware_metrics.rs

```diff
@@ -0,0 +1,280 @@
+                _ => vendor_id,
+            }
+        };
+
+        let cpu_model: &str = {
+            let brand = system
+                .cpus()
+                .first()
+                .map_or("cpu_model_unavailable", |cpu| cpu.brand());
+            match brand {
+                "" => "cpu_model_unavailable",
+                _ => brand,
+            }
+        };
+        let cpu_core_count = system.physical_core_count();
+
+        let mem_total_bytes = system.total_memory();
+        let memory_collector = IntGauge::with_opts(
+            Opts::new(
+                "memory_specs",
+                "Memory specs (constants: total amount, ...)",
+            )
+            .const_label("memory_total_bytes", mem_total_bytes.to_string())
+            .const_label("memory_total_human", human_fmt_bytes(mem_total_bytes)),
+        )
+        .map_err(HardwareMetricsErr::ErrCreateMetric)?;
+        memory_collector.set(mem_total_bytes as i64);
+
+        // We're only interested in the largest disk
+        let disks = Disks::new_with_refreshed_list();
+        let disk_total_bytes: u64 = disks
+            .iter()
+            .max_by_key(|disk| disk.total_space())
+            .map(|d| d.total_space())
+            .unwrap_or(0);
+        let disk_collector = IntGauge::with_opts(
+            Opts::new("disk_specs", "Disk specifications (total disk space, ...)")
+                .const_label("disk_total_bytes", disk_total_bytes.to_string())
+                .const_label("disk_total_space_human", human_fmt_bytes(disk_total_bytes)),
+        )
+        .map_err(HardwareMetricsErr::ErrCreateMetric)?;
+        disk_collector.set(disk_total_bytes as i64);
+
+        Ok(Self {
+            cpu_model: cpu_model.to_string(),
+            cpu_vendor_id: cpu_vendor_id.to_string(),
+            cpu_core_count,
+            memory_collector,
+            disk_collector,
+            is_docker: is_running_in_docker(),
+        })
+    }
+
+    fn label(name: &str, value: impl ToString) -> LabelPair {
+        let mut label = LabelPair::new();
+        label.set_name(name.to_string());
+        label.set_value(value.to_string());
+        label
+    }
+
+    fn cpu_metric(&self) -> Metric {
+        let mut metric = Metric::new();
+        metric.set_label({
+            vec![
+                Self::label("cpu_model", &self.cpu_model),
+                Self::label("cpu_vendor_id", &self.cpu_vendor_id),
+                Self::label(
+                    "cpu_core_count",
+                    self.cpu_core_count.map_or_else(
+                        || "cpu_core_count_unavailable".to_string(),
+                        |c| c.to_string(),
+                    ),
+                ),
+                Self::label("cpu_arch", System::cpu_arch()),
+            ]
+            .into()
+        });
+        metric
+    }
+    fn cpu_metric_family(&self) -> MetricFamily {
+        let mut mf = MetricFamily::new();
```

Comment:
Do we even need to recreate them during every collect? Those are all fixed values, can't we simply keep the final metric family in memory and reuse that?

Reply:
I can definitely do that. 
I had it at some point, but with const labels, which didn't work at all for displaying the data in grafana.